korma/src/commonMain/kotlin/com/soywiz/korma/interpolation/Easing.kt

```diff
@@ -2,92 +2,160 @@ package com.soywiz.korma.interpolation
 
 import kotlin.math.*
 
+private inline fun combine(it: Double, start: Easing, end: Easing) =
+    if (it < 0.5) 0.5 * start(it * 2.0) else 0.5 * end((it - 0.5) * 2.0) + 0.5
+private const val BOUNCE_FACTOR = 1.70158
+private const val HALF_PI = PI / 2.0
+
 @Suppress("unused")
 fun interface Easing {
     operator fun invoke(it: Double): Double
 
     companion object {
         fun cubic(f: (t: Double, b: Double, c: Double, d: Double) -> Double): Easing = Easing { f(it, 0.0, 1.0, 1.0) }
-        fun combine(start: Easing, end: Easing) = Easing { if (it < 0.5) 0.5 * start(it * 2.0) else 0.5 * end((it - 0.5) * 2.0) + 0.5 }
-
-        operator fun invoke(f: (Double) -> Double) = Easing { it -> f(it) }
+        fun combine(start: Easing, end: Easing) = Easing { combine(it, start, end) }
 
-        val SMOOTH get() = Easings.SMOOTH
-        val EASE_IN_ELASTIC get() = Easings.EASE_IN_ELASTIC
-        val EASE_OUT_ELASTIC get() = Easings.EASE_OUT_ELASTIC
-        val EASE_OUT_BOUNCE get() = Easings.EASE_OUT_BOUNCE
-        val LINEAR get() = Easings.LINEAR
-        val EASE_IN get() = Easings.EASE_IN
-        val EASE_OUT get() = Easings.EASE_OUT
-        val EASE_IN_OUT get() = Easings.EASE_IN_OUT
-        val EASE_OUT_IN get() = Easings.EASE_OUT_IN
-        val EASE_IN_BACK get() = Easings.EASE_IN_BACK
-        val EASE_OUT_BACK get() = Easings.EASE_OUT_BACK
-        val EASE_IN_OUT_BACK get() = Easings.EASE_IN_OUT_BACK
-        val EASE_OUT_IN_BACK get() = Easings.EASE_OUT_IN_BACK
-        val EASE_IN_OUT_ELASTIC get() = Easings.EASE_IN_OUT_ELASTIC
-        val EASE_OUT_IN_ELASTIC get() = Easings.EASE_OUT_IN_ELASTIC
-        val EASE_IN_BOUNCE get() = Easings.EASE_IN_BOUNCE
-        val EASE_IN_OUT_BOUNCE get() = Easings.EASE_IN_OUT_BOUNCE
-        val EASE_OUT_IN_BOUNCE get() = Easings.EASE_OUT_IN_BOUNCE
-        val EASE_IN_QUAD get() = Easings.EASE_IN_QUAD
-        val EASE_OUT_QUAD get() = Easings.EASE_OUT_QUAD
-        val EASE_IN_OUT_QUAD get() = Easings.EASE_IN_OUT_QUAD
-        val EASE_SINE get() = Easings.EASE_SINE
-    }
-}
+        operator fun invoke(f: (Double) -> Double) = Easing { f(it) }
```

Comment:
This method should not be required anymore since changed `interface Easing` into `fun interface Easing`, which is why I suggested to change enum members to `SOMETHING(Easing { })`. Please check this: https://kotlinlang.org/docs/reference/whatsnew14.html#sam-conversions-for-kotlin-interfaces